### DISCUSSION TITLE
What is the preferred way to programatically stop a cordova application in ios ?

### DISCUSSION THREAD

**Discussion by ghevge** [apache/cordova#406] [Q&A]
I'm looking for a way to programatically kill my app when running in ios. Is there a way to achieve this in cordova?

Thanks.

**Comment by breautek** [comment]
~There is a `navigator.app.exitApp` however I'm not sure if iOS supports that method. It's generally used by the android platform to handle the hardware back button.~

`navigator.app.exitApp` does appear to be an android only API, my bad.

**Reply by breautek** [reply]
From what I can gather, iOS SDK doesn't provide a way to gracefully exit an application, and it's not recommended by Apple either.

I couldn't find anything in the current docs, but I did find [this](https://developer.apple.com/library/archive/qa/qa1561/_index.html) in their archives.

iOS does have the standard `exit` native call, which will forcefully kill the app. This API is not exposed by Cordova. Calling on the native `exit` will not run through the normal exit routines and it will immediately exit the process, resulting in "crash-like" behaviour. It will not wait for thread suspensions or background tasks from completing it's work, so it can be dangerous to call.
